slitlessutils/core/sources/sourcecollection.py

```diff
@@ -124,10 +124,11 @@ def __init__(self, segfile, detfile, maglim=np.inf, minpix=0, preprocessor=None,
                 LOGGER.warning(msg)
                 return
 
-            # get the ZEROPOINT and THROUGHPUT
-            self.zeropoint = self.get_zeropoint(hdud, zeropoint=zeropoint)
-            self.throughput = self.get_throughput(hdud, throughput=throughput,
-                                                  **kwargs)
+            # get the throughput curve
+            self.throughput = self.get_throughput(hdud, throughput, **kwargs)
+
+            # set the zeropoint to the throughput
+            self.throughput.zeropoint = self.get_zeropoint(hdud, zeropoint)
```

Comment:
```suggestion
            self.throughput.zeropoint = self.get_zeropoint(hdud) if zeropoint is None else zeropoint  # noqa
```
Rather than passing a non-`None` zeropoint into this function just to return it, how about a case here to just not call the function if it's not needed? Seems cleaner. Similarly, don't need to call `self.get_throughput` if `isinstance(throughput, Throughput)` is already satisfied. (Note that the `# noqa` is because the suggestion is 101 characters long and I didn't feel like splitting it up...)

Reply:
Yeah, I see what you mean.  I did it this way because I wanted to be able to catch all type conditions someone might pass and this is a relic of transcribing IDL code.  For example, if someone passes zero point as a ``str``, then the ternary condition you suggested will pass (i mean a ``str`` is not None).  Therefore, it will assign ``self.zeropoint`` as a string.  of course, this isn't a big deal, as later the addition operator would catch the problem, when you do something like:

```
mag = -2.5*np.log10(flux) + self.zeropoint
```

but it gives a fairly innocuous error message:
```
 File "<stdin>", line 1, in <module>
numpy.core._exceptions._UFuncNoLoopError: ufunc 'add' did not contain a loop with signature matching types (dtype('float64'), dtype('<U3')) -> None
```

And might be hard for user to decipher what they'd done wrong.  So, I thought to explicitly catch this error.  I did this, because again, it was largely transcribed from IDL, where if you add a string to a float, it can give unexpected behaviors (depending on the version).  

So, unless you feel strongly, I'd prefer to keep the validation of the variable inside the helper function so it's consolidated.  That said, I think I should (1) update the warning message to reflect the intentions, and (2) maybe rename the functions ``get_zeropoint`` and ``get_throughput`` to ``_get_zeropoint`` and ``_get_throughput`` to make it a bit clearer that these are unlikely things that one should usually interact with.

Reply:
Whoops, I think I lost my train of thought halfway through writing this suggestion. As I said offline, the suggestion should have been 

`self.throughput.zeropoint = self.get_zeropoint(hdud) if not isinstance(zeropoint, (int, float))  else zeropoint  # noqa`

since `(int, float)` is the case where it just returns the input, not `None`.